---
title: "Main analysis Script"
author: "Emil Hvitfeldt"
date: "2019-09-15"
output: md_document
params:
  prep: FALSE
---

```{r setup, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(cache = TRUE, echo = FALSE)
library(tidyverse)
library(here)
library(gt)
library(rlang)
library(minfi)

select <- dplyr::select
rename <- dplyr::rename
filter <- dplyr::filter

dir.create(here("figures"))

set.seed(1234)
```

The **methcon5** package have been updated since this writeup. Please install old version to complete analysis

```{r}
#remotes::install_github("EmilHvitfeldt/methcon5@0.0.0.9000")
library(methcon5)
```

## Data Preparation

The following function, will take a folder containing .idat files and calculates the noob adjusted beta values for the samples.

```{r , echo = TRUE, eval=params$prep}
idat_to_beta_values_noob <- function(base) {
  RGset <- read.metharray.exp(base = base)
  RGset@annotation = c(array = "IlluminaHumanMethylationEPIC", annotation = "ilm10b4.hg19")

  MSet.noob <- preprocessNoob(RGset, offset = 15, dyeCorr = TRUE, verbose = TRUE)

  ratioSet.noob <- ratioConvert(MSet.noob, what =  "both", keepCN = TRUE)
  beta.noob <- getBeta(ratioSet.noob)
  beta.noob
}
```

This function is applied to each of the tissue folders. 

```{r idatfiles, echo = TRUE, eval=params$prep}
colon_idat <- idat_to_beta_values_noob(base = "data-raw/colon-idat/")
si_idat <- idat_to_beta_values_noob(base = "data-raw/si-idat/")
endo_idat <- idat_to_beta_values_noob(base = "data-raw/endo-idat/")
```

Infinium MethylationEPIC v1.0 B4 Manifest File (CSV Format) is to be downloaded from the following link
https://support.illumina.com/downloads/infinium-methylationepic-v1-0-product-files.html and `MethylationEPIC_v-1-0_B4.csv` should be placed in the folder `data-raw`. `pwd()` will take a vector and calculate the average manhattan distance. We then apply it row-wise to the methylation data. Lastly we take the calulated average manhattan distances and append  it to the manifest file. 

```{r prepped_data, echo = TRUE, eval=params$prep}
pwd <- function(x) {
  mean(dist(x, method = "manhattan"))
}

colon_pwd_values <- colon_idat %>% apply(1, pwd)
si_pwd_values <- si_idat %>% apply(1, pwd)
endo_pwd_values <- endo_idat %>% apply(1, pwd)

prepped_data <- 
  read_csv(here("data-raw/MethylationEPIC_v-1-0_B4.csv"), skip = 7, 
                     col_types = cols(Name = col_character(),
                                      CHR = col_character(),
                                      MAPINFO = col_double(),
                                      UCSC_RefGene_Name = col_character(),
                                      UCSC_RefGene_Group = col_character(),
                                      Relation_to_UCSC_CpG_Island = col_character())) %>%
  left_join(by = "Name",
            tibble(Name = names(colon_pwd_values),
                   colon_pwd = colon_pwd_values)) %>%
  left_join(by = "Name",
            tibble(Name = names(si_pwd_values),
                   si_pwd = si_pwd_values)) %>%
  left_join(by = "Name",
            tibble(Name = names(endo_pwd_values),
                   endo_pwd = endo_pwd_values)) %>%
  select(-IlmnID, -(AddressA_ID:Genome_Build), -(SourceSeq:Strand), -UCSC_RefGene_Accession, 
         -UCSC_CpG_Islands_Name, -Phantom4_Enhancers, -(DMR:X48), -Phantom5_Enhancers)

write_csv(prepped_data, here("data", "prepped_data.csv"))
```

Next we seperate the rows to apply for a CpG site to appear in multiple genes as annotated.

```{r gened_data, echo = TRUE, eval=params$prep}
gened_data <- prepped_data %>%
  arrange(CHR, MAPINFO) %>%
  mutate(temp = floor(row_number() / n() * 100)) %>%
  separate_rows(UCSC_RefGene_Name) %>% 
  distinct()

write_csv(gened_data, here("data", "gened_data.csv"))
```

Here we apply the the first function from methcon5, we calculates the unadjusted conservation for each gene for each of the tissue types.

```{r pwd_by_gene, echo = TRUE, eval=params$prep}
gened_data <- read_csv(here("data", "gened_data.csv"))
pwd_by_gene <- gened_data %>%
  filter(!is.na(UCSC_RefGene_Name)) %>%
  ii_summarize(UCSC_RefGene_Name, colon_pwd:endo_pwd)
```

Next we apply the second function from methcon5 where we will apply two different boot-strapping method to each of the tissuetypes.

```{r permed_data, echo = TRUE, eval=params$prep}
permed_data <- pwd_by_gene %>%
  filter(n < quantile(n, 0.99)) %>%
  bootter(gened_data, colon_pwd, UCSC_RefGene_Name, 1000,  method = 1) %>%
  bootter(gened_data, si_pwd,    UCSC_RefGene_Name, 1000, method = 1) %>%
  bootter(gened_data, endo_pwd,  UCSC_RefGene_Name, 1000, method = 1) %>%
  bootter(gened_data, colon_pwd, UCSC_RefGene_Name, 1000,  method = 3) %>%
  bootter(gened_data, si_pwd,    UCSC_RefGene_Name, 1000, method = 3) %>%
  bootter(gened_data, endo_pwd,  UCSC_RefGene_Name, 1000, method = 3)

write_csv(permed_data, here("data", "permed_data.csv"))
```

```{r loadingdata, eval=!params$prep}
gened_data <- read_csv(here("data", "gened_data.csv"))
permed_data <- read_csv(here("data", "permed_data.csv"))
prepped_data <- read_csv(here("data", "prepped_data.csv"))
```

Next we calculate the mean methylation for CpG sites in the promotor region.

```{r promoted_data, echo = TRUE}
promoted_data <- prepped_data %>%
  mutate(gene = str_extract(UCSC_RefGene_Name, "^[^;]+")) %>%
  filter(!is.na(gene)) %>%
  left_join(read_csv(here("data-raw/MethylationEPIC_v-1-0_B4.csv"), skip = 7) %>%  
              dplyr::select(Name, Regulatory_Feature_Group), 
            by = "Name") %>%
  filter(Regulatory_Feature_Group == "Promoter_Associated") %>%
  group_by(gene) %>%
  summarise(colon_pro = mean(colon_pwd),
            si_pro = mean(si_pwd),
            endo_pro = mean(endo_pwd))
```

Expression data can be  downloaded from the following website https://www.ebi.ac.uk/gxa/experiments/E-MTAB-2836/Results, and the resulting file `E-MTAB-2836-query-results.tpms.tsv` should be placed in the folder named `data-raw`.

```{r expression_atlas, echo = TRUE}
expression_atlas <- read_tsv(here("data-raw", "E-MTAB-2836-query-results.tpms.tsv"), skip = 4, 
                             col_types = cols(
  .default = col_double(),
  `Gene ID` = col_character(),
  `Gene Name` = col_character()
))
```

## Gene enrichment set analysis

In the following code we apply gene enrichment set analysis on the 5% most conserved genes for each tissue type one at a time.

```{r gesa, echo = TRUE}
tidy_gsea <- function(genes) {
  AnnotationDbi::mapIds(org.Hs.eg.db::org.Hs.eg.db, genes, 'ENTREZID', 'SYMBOL') %>%
  ReactomePA::enrichPathway()
}

enrich_colon <- permed_data %>%
  filter(colon_pwd_v3 > quantile(colon_pwd_v3, 0.95, na.rm = TRUE)) %>%
  pull(UCSC_RefGene_Name) %>%
  tidy_gsea()

enrich_si <- permed_data %>%
  filter(si_pwd_v3 > quantile(si_pwd_v3, 0.95, na.rm = TRUE)) %>%
  pull(UCSC_RefGene_Name) %>%
  tidy_gsea()

enrich_endo <- permed_data %>%
  filter(endo_pwd_v3 > quantile(endo_pwd_v3, 0.95, na.rm = TRUE)) %>%
  pull(UCSC_RefGene_Name) %>%
  tidy_gsea()
```

## Table 1

The calculations happening in the following bits of code follow the same steps. Split the data accoding to a specified region. Then calculate the mean pwd for each tissue type.

```{r table1_gene, echo = TRUE}
prepped_data %>%
  group_by(`Gene Associated` = !is.na(UCSC_RefGene_Name)) %>%
  summarise_at(vars(colon_pwd:endo_pwd), mean, na.rm = TRUE) %>%
  rename(Colon = colon_pwd,
         `Small Intestine` = si_pwd,
         `Endometrium` = endo_pwd) %>%
  mutate(`Gene Associated` = factor(`Gene Associated`, c(TRUE, FALSE), c("Yes", "No"))) %>%
  arrange(rev(rownames(.))) %>%
  gt() %>%
  fmt_number(columns = vars(Colon, `Small Intestine`, `Endometrium`), decimals = 3) %>%
  tab_spanner(
    label = "Manhattan distance",
    columns = vars(Colon, `Small Intestine`, `Endometrium`)
  )
```

```{r table1_island, echo = TRUE}
prepped_data %>%
  group_by(`Island relation` = Relation_to_UCSC_CpG_Island) %>%
  summarise_at(vars(colon_pwd:endo_pwd), mean, na.rm = TRUE) %>%
  rename(Colon = colon_pwd,
         `Small Intestine` = si_pwd,
         `Endometrium` = endo_pwd) %>%
  dplyr::rename(ir = `Island relation`) %>%
  mutate(ir = case_when(is.na(ir) ~ "Sea",
                        str_detect(ir, "^S") ~ str_replace(ir, "^S_", "South "),
                        str_detect(ir, "^N") ~ str_replace(ir, "^N_", "North "),
                        TRUE ~ ir)) %>%
  dplyr::rename(`Island relation` = ir) %>%
  arrange(rev(rownames(.))) %>%
  gt() %>%
  fmt_number(columns = vars(Colon, `Small Intestine`, `Endometrium`), decimals = 3) %>%
  tab_spanner(
    label = "Manhattan distance",
    columns = vars(Colon, `Small Intestine`, `Endometrium`)
  )
```

```{r table1_5utr, echo = TRUE}
prepped_data %>%
  group_by(`5'UTR` = str_detect(UCSC_RefGene_Group, "5'UTR") & !is.na(UCSC_RefGene_Group)) %>%
  summarise_at(vars(colon_pwd:endo_pwd), mean, na.rm = TRUE) %>%
  rename(Colon = colon_pwd,
         `Small Intestine` = si_pwd,
         `Endometrium` = endo_pwd) %>%
  mutate(`5'UTR` = factor(`5'UTR`, c(TRUE, FALSE), c("Yes", "No"))) %>%
  arrange(rev(rownames(.))) %>%
  gt() %>%
  fmt_number(columns = vars(Colon, `Small Intestine`, `Endometrium`), decimals = 3) %>%
  tab_spanner(
    label = "Manhattan distance",
    columns = vars(Colon, `Small Intestine`, `Endometrium`)
  )
```

```{r table1tss1500, echo = TRUE}
prepped_data %>%
  group_by(`TSS1500` = str_detect(UCSC_RefGene_Group, "TSS1500") & !is.na(UCSC_RefGene_Group)) %>%
  summarise_at(vars(colon_pwd:endo_pwd), mean, na.rm = TRUE) %>%
  rename(Colon = colon_pwd,
         `Small Intestine` = si_pwd,
         `Endometrium` = endo_pwd) %>%
  mutate(`TSS1500` = factor(`TSS1500`, c(TRUE, FALSE), c("Yes", "No"))) %>%
  arrange(rev(rownames(.))) %>%
  gt() %>%
  fmt_number(columns = vars(Colon, `Small Intestine`, `Endometrium`), decimals = 3) %>%
  tab_spanner(
    label = "Manhattan distance",
    columns = vars(Colon, `Small Intestine`, `Endometrium`)
  )
```

```{r table1tss200, echo = TRUE}
prepped_data %>%
  group_by(`TSS200` = str_detect(UCSC_RefGene_Group, "TSS200") & !is.na(UCSC_RefGene_Group)) %>%
  summarise_at(vars(colon_pwd:endo_pwd), mean, na.rm = TRUE) %>%
  rename(Colon = colon_pwd,
         `Small Intestine` = si_pwd,
         `Endometrium` = endo_pwd) %>%
  mutate(`TSS200` = factor(`TSS200`, c(TRUE, FALSE), c("Yes", "No"))) %>%
  arrange(rev(rownames(.))) %>%
  gt() %>%
  fmt_number(columns = vars(Colon, `Small Intestine`, `Endometrium`), decimals = 3) %>%
  tab_spanner(
    label = "Manhattan distance",
    columns = vars(Colon, `Small Intestine`, `Endometrium`)
  )
```

## Figure 1

```{r distance_length, echo = TRUE}
distance_length <- gened_data %>%
  filter(!is.na(UCSC_RefGene_Name)) %>%
  select(CHR, MAPINFO, UCSC_RefGene_Name, colon_pwd:endo_pwd) %>%
  group_by(UCSC_RefGene_Name, CHR) %>%
  mutate(location = MAPINFO - MAPINFO[1]) %>%
  mutate(location_bin = cut(location, c(-1, seq(0, 20000, by = 100)), include.lowest = TRUE)) %>%
  group_by(location_bin) %>%
  summarise(mean_colon = mean(colon_pwd),
            mean_si = mean(si_pwd),
            mean_endo = mean(endo_pwd),
            n_obs = n()) %>%
  mutate(location_bin = as.numeric(str_extract(location_bin, "(?<=,)(.*)(?=])")))
```

```{r figure1, echo = TRUE}
distance_length %>%
  drop_na() %>%
  pivot_longer(mean_colon:mean_endo) %>% 
  mutate(name = factor(name, c("mean_colon", "mean_si", "mean_endo"),
                             c("Colon", "Small Intestine", "Endometrium"))) %>%
  sample_frac() %>%
  ggplot(aes(location_bin, value, color = name, fill = name)) +
  geom_point(key_glyph = draw_key_rect) +
  ggforce::facet_zoom(location_bin < 5000) +
  theme_light() +
  theme(legend.position = "bottom",
        plot.caption = element_text(hjust = 0)) +
  labs(title = NULL,
       fill = NULL,
       color = NULL,
       x = "Distance by hg19 position from first cpg site (binned to 100s)",
       y = "Average Manhattan Distance", 
       caption = str_wrap("Figure 1: Average Manhattan distance for single CpGs as a function of position relative to first 5' annotated gene CpG site.  The greater conservation (lower average Manhattan distances) around genes indicates DNA methylation conservation generally extends for hundreds of base pairs and is not isolated to a single CpG site.", 100)) +
  scale_color_brewer(palette = "Set2") +
  scale_fill_brewer(palette = "Set2") 

ggsave(filename = here("figures", "figure-1.png"), width = 7, dpi = 320, height = 4.326)
```

## Figure 2

```{r figure2, echo = TRUE}
permed_data %>%
  pivot_longer(dplyr::matches("1|3")) %>%
  separate(name, into = c("type", "method"), sep = "_v") %>%
  mutate(method = factor(method, c("1", "3"),
                             c("Naive bootstrap",
                               "Adjusted bootstrap")),
         type = factor(type, c("colon_pwd", "si_pwd", "endo_pwd"),
                             c("Colon", "Small Intestine", "Endometrium"))) %>%
  ggplot(aes(value, fill = type)) +
  geom_histogram(bins = 50, color = "grey30") +
  facet_grid(method~type, scales = "free") +
  theme_minimal() +
  theme(plot.caption = element_text(hjust = 0)) +
  labs(x = "p-value",
       y = "Number of genes",
       title = NULL,
       caption = str_wrap("Figure 2: Distribution of boot-strapping p-values for genes. Each column corresponds to a specific tissue. The top row shows results from the naive bootstrap procedure, whereas the bottom row shows the adjusted bootstrap results."), 100)  +
  theme(legend.position = "none") +
  scale_color_brewer(palette = "Set2") +
  scale_fill_brewer(palette = "Set2")

ggsave(filename = here("figures", "figure-2.png"), width = 7, dpi = 320, height = 4.326)
```

## Figure 3

```{r figure3, echo = TRUE}

c(colon = length(enrich_colon$Description),
  si = length(enrich_si$Description),
  endo = length(enrich_endo$Description)) %>%
  sort()

list(
  Colon             = enrich_colon$Description,
  `Small Intestine` = enrich_si$Description,
  Endometrium       = enrich_endo$Description
) %>%
  UpSetR::fromList() %>%
  UpSetR::upset(sets.bar.color = RColorBrewer::brewer.pal(3, "Set2")[c(3, 2, 1)])
grid::grid.text("Figure 3: Frequency of overlap between pathways that are called \nas most conserved for each tissue type.",x = 0.65, y=0.05, gp=grid::gpar(fontsize=10))

png(file=here("figures", "figure-3.png"), width = 7, height = 4.326, units = "in", res = 320) # or other device
list(
  Colon             = enrich_colon$Description,
  `Small Intestine` = enrich_si$Description,
  Endometrium       = enrich_endo$Description
) %>%
  UpSetR::fromList() %>%
  UpSetR::upset(sets.bar.color = RColorBrewer::brewer.pal(3, "Set2")[c(3, 2, 1)])
dev.off()
```

## Figure 4

```{r figure4, echo = TRUE}
ggg <- enrich_si %>%
  ReactomePA::emapplot(showCategory = 30, layout = "graphopt") +
  labs(caption = "Figure 4: Relationship between pathways that are called as conserved in small intestine tissue.",
       title = NULL)

ggg$layers <- ggg$layers[1:2]

ggg + 
  ggraph::geom_node_label(aes_(label = ~name), repel = TRUE, size = 2) +
  scale_color_gradientn(colors = "black") +
  guides(color = "none",
         size = "none")

ggsave(filename = here("figures", "figure-4.png"), width = 7, dpi = 320, height = 4.326)
```

Interactive enrichment map 

```{r}
n <- enrichplot:::update_n(enrich_si, 30)
gene_sets <- DOSE::geneInCategory(enrich_si)
y <- as.data.frame(enrich_si)


id <- y[, 1]
gene_sets <- gene_sets[id]
n <- nrow(y)
w <- matrix(NA, nrow = n, ncol = n)
colnames(w) <- rownames(w) <- y$Description
for (i in 1:n) {
    for (j in i:n) {
        w[i, j] = enrichplot:::overlap_ratio(gene_sets[id[i]], gene_sets[id[j]])
    }
}

wd <- reshape2::melt(w)
wd <- wd[wd[, 1] != wd[, 2], ]
wd <- wd[!is.na(wd[, 3]), ]

g <- igraph::graph.data.frame(wd[, -3], directed = FALSE)
igraph::E(g)$width = sqrt(wd[, 3] * 5)
g <- igraph::delete.edges(g, igraph::E(g)[wd[, 3] < 0.5])
idx <- unlist(sapply(igraph::V(g)$name, function(x) which(x == 
    y$Description)))
cnt <- sapply(gene_sets[idx], length)
igraph::V(g)$size <- cnt
colVar <- y[idx, "p.adjust"]
igraph::V(g)$color <- colVar

library(networkD3)
library(igraph)

wc <- cluster_walktrap(g)
members <- membership(wc)

d3 <- igraph_to_networkD3(g, group = members)

# Create force directed network plot
forceNetwork(Links = d3$links, Nodes = d3$nodes, 
             Source = 'source', Target = 'target', 
             NodeID = 'name', Group = 'group', charge = -5, bounded = TRUE, opacity = 1, fontSize = 14)
```

```{r}

topic_summary <- function(x) {
  x %>%
    names() %>%
    tokenizers::tokenize_words(stopwords = stopwords::stopwords(source = "stopwords-iso")) %>%
    unlist() %>%
    table() %>%
    sort(decreasing = TRUE) %>%
    .[1:3] %>%
    names() %>%
    paste(collapse = ", ")
}

```

```{r}
# blue
members[members == 1] %>%
  topic_summary()

# light green
members[members == 2] %>%
  topic_summary()

# beige
members[members == 3] %>%
  topic_summary()

# Red
members[members == 4] %>%
  topic_summary()

# light blue
members[members == 5] %>%
  topic_summary()

# purple
members[members == 7] %>%
  topic_summary()
```

## Figure 5

```{r figure5, echo = TRUE}
figure5_data <- permed_data %>% 
  left_join(by = c("UCSC_RefGene_Name" = "gene"),
  promoted_data
  ) %>%
  left_join(expression_atlas %>% dplyr::select(`Gene Name`, colon, endometrium, `small intestine`),
            by = c("UCSC_RefGene_Name" = "Gene Name")) 

bind_rows(
  figure5_data %>%
  select(colon_pwd_v3, colon_pro, colon) %>%
  pivot_longer(-colon) %>%
  drop_na() %>%
  rename(expression = colon),
  figure5_data %>%
  select(si_pwd_v3, si_pro, `small intestine`) %>%
  pivot_longer(-`small intestine`) %>%
  drop_na() %>%
  rename(expression = `small intestine`),
  figure5_data %>%
  select(endo_pwd_v3, endo_pro, endometrium) %>%
  pivot_longer(-endometrium) %>%
  drop_na() %>%
  rename(expression = endometrium)
) %>%
  group_by(name) %>%
  mutate(value = factor(as.numeric(cut_number(value, 10)), levels = 1:10, 
                        labels = paste0(0:9 * 10, "-", 1:10 * 10, "%"))) %>%
  separate(name, c("type", "method"), sep = "_", extra = "merge") %>%
  mutate(type = factor(type, c("colon", "si", "endo"), c("Colon", "Small Intenstine", "Endometrium")),
         method = factor(method, c("pwd_v3", "pro"), c("Boot-strapped values", "Promoter region"))) %>%
  ggplot(aes(value, expression, fill = type)) +
  geom_boxplot() +
  scale_y_log10() +
  guides(color = "none") +
  facet_grid(method ~ type, scales = "free_x") +
  labs(x = "quantile of conservation by gene", 
       y = "Expression",
       title = NULL,
       caption = str_wrap("Figure 5: The relationship between conservation and expression. Genes are collected into 10 groups according to the degree of conservation measured in our data. For each group, we then show a box-pot of the distribution of log(gene expression) values recorded for the corresponding tissue type in the Expression Atlas database. Columns correspond to the tissue type. The top row shows results when assessing conservation for the entire gene; the bottom row shows the results when assessing conservation just for the promoter region of each gene. We see that gene conservation correlates with expression better than does promoter conservation.", 100)) +
  theme_minimal() +
  theme(plot.caption = element_text(hjust = 0)) +
  scale_fill_brewer(palette = "Set2") +
  theme(legend.position = "none", 
        axis.text.x = element_text(angle = 90, hjust = 1))

ggsave(filename = here("figures", "figure-5.png"), width = 7, dpi = 320, height = 4.326)
```

```{r}
bind_rows(
  figure5_data %>%
  select(colon_pwd_v3, colon_pro, colon) %>%
  pivot_longer(-colon) %>%
  drop_na() %>%
  rename(expression = colon),
  figure5_data %>%
  select(si_pwd_v3, si_pro, `small intestine`) %>%
  pivot_longer(-`small intestine`) %>%
  drop_na() %>%
  rename(expression = `small intestine`),
  figure5_data %>%
  select(endo_pwd_v3, endo_pro, endometrium) %>%
  pivot_longer(-endometrium) %>%
  drop_na() %>%
  rename(expression = endometrium)
) %>%
  ggplot(aes(value)) +
  geom_histogram()
```

