---
title: "Main analysis Script"
author: "Emil Hvitfeldt"
date: "2019-09-15"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(cache = TRUE, echo = FALSE)
library(tidyverse)
library(here)
library(gt)
library(rlang)
library(methcon5)
library(minfi)

select <- dplyr::select
rename <- dplyr::rename
filter <- dplyr::filter

dir.create(here("figures"))

set.seed(1234)
```

## Data Preparation

```{r}
idat_to_beta_values_noob <- function(base) {
  RGset <- read.metharray.exp(base = base)
  RGset@annotation = c(array = "IlluminaHumanMethylationEPIC", annotation = "ilm10b4.hg19")

  MSet.noob <- preprocessNoob(RGset, offset = 15, dyeCorr = TRUE, verbose = TRUE)

  ratioSet.noob <- ratioConvert(MSet.noob, what =  "both", keepCN = TRUE)
  beta.noob <- getBeta(ratioSet.noob)
  beta.noob
}
```

```{r idatfiles}
colon_idat <- idat_to_beta_values_noob(base = "data-raw/colon-idat/")
si_idat <- idat_to_beta_values_noob(base = "data-raw/si-idat/")
endo_idat <- idat_to_beta_values_noob(base = "data-raw/endo-idat/")
```

```{r prepped_data}
pwd <- function(x) {
  mean(dist(x, method = "manhattan"))
}

colon_pwd_values <- colon_idat %>% apply(1, pwd)
si_pwd_values <- si_idat %>% apply(1, pwd)
endo_pwd_values <- endo_idat %>% apply(1, pwd)

prepped_data <- 
  read_csv(here("data-raw/MethylationEPIC_v-1-0_B4.csv"), skip = 7, 
                     col_types = cols(Name = col_character(),
                                      CHR = col_character(),
                                      MAPINFO = col_double(),
                                      UCSC_RefGene_Name = col_character(),
                                      UCSC_RefGene_Group = col_character(),
                                      Relation_to_UCSC_CpG_Island = col_character())) %>%
  left_join(by = "Name",
            tibble(Name = names(colon_pwd_values),
                   colon_pwd = colon_pwd_values)) %>%
  left_join(by = "Name",
            tibble(Name = names(si_pwd_values),
                   si_pwd = si_pwd_values)) %>%
  left_join(by = "Name",
            tibble(Name = names(endo_pwd_values),
                   endo_pwd = endo_pwd_values)) %>%
  select(-IlmnID, -(AddressA_ID:Genome_Build), -(SourceSeq:Strand), -UCSC_RefGene_Accession, 
         -UCSC_CpG_Islands_Name, -Phantom4_Enhancers, -(DMR:X48), -Phantom5_Enhancers)

write_csv(prepped_data, here("data", "prepped_data.csv"))
```

```{r gened_data}
gened_data <- prepped_data %>%
  arrange(CHR, MAPINFO) %>%
  mutate(temp = floor(row_number() / n() * 100)) %>%
  separate_rows(UCSC_RefGene_Name) %>% 
  distinct()

write_csv(gened_data, here("data", "gened_data.csv"))
```

```{r pwd_by_gene}
gened_data <- read_csv(here("data", "gened_data.csv"))
pwd_by_gene <- gened_data %>%
  filter(!is.na(UCSC_RefGene_Name)) %>%
  ii_summarize(UCSC_RefGene_Name, colon_pwd:endo_pwd)
```

```{r permed_data}
permed_data <- pwd_by_gene %>%
  filter(n < quantile(n, 0.99)) %>%
  bootter(gened_data, colon_pwd, UCSC_RefGene_Name, 1000,  method = 1) %>%
  bootter(gened_data, si_pwd,    UCSC_RefGene_Name, 1000, method = 1) %>%
  bootter(gened_data, endo_pwd,  UCSC_RefGene_Name, 1000, method = 1) %>%
  bootter(gened_data, colon_pwd, UCSC_RefGene_Name, 1000,  method = 3) %>%
  bootter(gened_data, si_pwd,    UCSC_RefGene_Name, 1000, method = 3) %>%
  bootter(gened_data, endo_pwd,  UCSC_RefGene_Name, 1000, method = 3)

write_csv(permed_data, here("data", "permed_data.csv"))
```

```{r promoted_data}
promoted_data <- prepped_data %>%
  mutate(gene = str_extract(UCSC_RefGene_Name, "^[^;]+")) %>%
  filter(!is.na(gene)) %>%
  left_join(read_csv(here("data-raw/MethylationEPIC_v-1-0_B4.csv"), skip = 7) %>%  
              dplyr::select(Name, Regulatory_Feature_Group), 
            by = "Name") %>%
  filter(Regulatory_Feature_Group == "Promoter_Associated") %>%
  group_by(gene) %>%
  summarise(colon_pro = mean(colon_pwd),
            si_pro = mean(si_pwd),
            endo_pro = mean(endo_pwd))
```

```{r expression_atlas}
expression_atlas <- read_tsv(here("data-raw", "E-MTAB-2836-query-results.tpms.tsv"), skip = 4, 
                             col_types = cols(
  .default = col_double(),
  `Gene ID` = col_character(),
  `Gene Name` = col_character()
))
```


## Gene enrichment set analysis

```{r gesa}
tidy_gsea <- function(genes) {
  AnnotationDbi::mapIds(org.Hs.eg.db::org.Hs.eg.db, genes, 'ENTREZID', 'SYMBOL') %>%
  ReactomePA::enrichPathway()
}

enrich_colon <- permed_data %>%
  filter(colon_pwd_v3 > quantile(colon_pwd_v3, 0.95, na.rm = TRUE)) %>%
  pull(UCSC_RefGene_Name) %>%
  tidy_gsea()

enrich_si <- permed_data %>%
  filter(si_pwd_v3 > quantile(si_pwd_v3, 0.95, na.rm = TRUE)) %>%
  pull(UCSC_RefGene_Name) %>%
  tidy_gsea()

enrich_endo <- permed_data %>%
  filter(endo_pwd_v3 > quantile(endo_pwd_v3, 0.95, na.rm = TRUE)) %>%
  pull(UCSC_RefGene_Name) %>%
  tidy_gsea()
```


## Table 1

```{r table1_gene}
prepped_data %>%
  group_by(`Gene Associated` = !is.na(UCSC_RefGene_Name)) %>%
  summarise_at(vars(colon_pwd:endo_pwd), mean, na.rm = TRUE) %>%
  rename(Colon = colon_pwd,
         `Small Intestine` = si_pwd,
         `Endometrium` = endo_pwd) %>%
  mutate(`Gene Associated` = factor(`Gene Associated`, c(TRUE, FALSE), c("Yes", "No"))) %>%
  arrange(rev(rownames(.))) %>%
  gt() %>%
  fmt_number(columns = vars(Colon, `Small Intestine`, `Endometrium`), decimals = 3) %>%
  tab_spanner(
    label = "Manhattan distance",
    columns = vars(Colon, `Small Intestine`, `Endometrium`)
  )
```

```{r table1_island}
prepped_data %>%
  group_by(`Island relation` = Relation_to_UCSC_CpG_Island) %>%
  summarise_at(vars(colon_pwd:endo_pwd), mean, na.rm = TRUE) %>%
  rename(Colon = colon_pwd,
         `Small Intestine` = si_pwd,
         `Endometrium` = endo_pwd) %>%
  dplyr::rename(ir = `Island relation`) %>%
  mutate(ir = case_when(is.na(ir) ~ "Sea",
                        str_detect(ir, "^S") ~ str_replace(ir, "^S_", "South "),
                        str_detect(ir, "^N") ~ str_replace(ir, "^N_", "North "),
                        TRUE ~ ir)) %>%
  dplyr::rename(`Island relation` = ir) %>%
  arrange(rev(rownames(.))) %>%
  gt() %>%
  fmt_number(columns = vars(Colon, `Small Intestine`, `Endometrium`), decimals = 3) %>%
  tab_spanner(
    label = "Manhattan distance",
    columns = vars(Colon, `Small Intestine`, `Endometrium`)
  )
```

```{r table1_5utr}
prepped_data %>%
  group_by(`5'UTR` = str_detect(UCSC_RefGene_Group, "5'UTR") & !is.na(UCSC_RefGene_Group)) %>%
  summarise_at(vars(colon_pwd:endo_pwd), mean, na.rm = TRUE) %>%
  rename(Colon = colon_pwd,
         `Small Intestine` = si_pwd,
         `Endometrium` = endo_pwd) %>%
  mutate(`5'UTR` = factor(`5'UTR`, c(TRUE, FALSE), c("Yes", "No"))) %>%
  arrange(rev(rownames(.))) %>%
  gt() %>%
  fmt_number(columns = vars(Colon, `Small Intestine`, `Endometrium`), decimals = 3) %>%
  tab_spanner(
    label = "Manhattan distance",
    columns = vars(Colon, `Small Intestine`, `Endometrium`)
  )
```

```{r table1tss1500}
prepped_data %>%
  group_by(`TSS1500` = str_detect(UCSC_RefGene_Group, "TSS1500") & !is.na(UCSC_RefGene_Group)) %>%
  summarise_at(vars(colon_pwd:endo_pwd), mean, na.rm = TRUE) %>%
  rename(Colon = colon_pwd,
         `Small Intestine` = si_pwd,
         `Endometrium` = endo_pwd) %>%
  mutate(`TSS1500` = factor(`TSS1500`, c(TRUE, FALSE), c("Yes", "No"))) %>%
  arrange(rev(rownames(.))) %>%
  gt() %>%
  fmt_number(columns = vars(Colon, `Small Intestine`, `Endometrium`), decimals = 3) %>%
  tab_spanner(
    label = "Manhattan distance",
    columns = vars(Colon, `Small Intestine`, `Endometrium`)
  )
```

```{r table1tss200}
prepped_data %>%
  group_by(`TSS200` = str_detect(UCSC_RefGene_Group, "TSS200") & !is.na(UCSC_RefGene_Group)) %>%
  summarise_at(vars(colon_pwd:endo_pwd), mean, na.rm = TRUE) %>%
  rename(Colon = colon_pwd,
         `Small Intestine` = si_pwd,
         `Endometrium` = endo_pwd) %>%
  mutate(`TSS200` = factor(`TSS200`, c(TRUE, FALSE), c("Yes", "No"))) %>%
  arrange(rev(rownames(.))) %>%
  gt() %>%
  fmt_number(columns = vars(Colon, `Small Intestine`, `Endometrium`), decimals = 3) %>%
  tab_spanner(
    label = "Manhattan distance",
    columns = vars(Colon, `Small Intestine`, `Endometrium`)
  )
```



## Figure 1

```{r distance_length}
distance_length <- gened_data %>%
  filter(!is.na(UCSC_RefGene_Name)) %>%
  select(CHR, MAPINFO, UCSC_RefGene_Name, colon_pwd:endo_pwd) %>%
  group_by(UCSC_RefGene_Name, CHR) %>%
  mutate(location = MAPINFO - MAPINFO[1]) %>%
  mutate(location_bin = cut(location, c(-1, seq(0, 20000, by = 100)), include.lowest = TRUE)) %>%
  group_by(location_bin) %>%
  summarise(mean_colon = mean(colon_pwd),
            mean_si = mean(si_pwd),
            mean_endo = mean(endo_pwd),
            n_obs = n()) %>%
  mutate(location_bin = as.numeric(str_extract(location_bin, "(?<=,)(.*)(?=])")))
```

```{r figure1}
distance_length %>%
  drop_na() %>%
  pivot_longer(mean_colon:mean_endo) %>% 
  mutate(name = factor(name, c("mean_colon", "mean_si", "mean_endo"),
                             c("Colon", "Small Intestine", "Endometrium"))) %>%
  sample_frac() %>%
  ggplot(aes(location_bin, value, color = name, fill = name)) +
  geom_point(key_glyph = draw_key_rect) +
  ggforce::facet_zoom(location_bin < 5000) +
  theme_light() +
  theme(legend.position = "bottom") +
  labs(title = "The lowest average Manhattan distance is located around 1500 basepairs \nfrom first CpG site in the gene",
       fill = NULL,
       color = NULL,
       x = "Distance by hg19 position from first cpg site (binned to 100s)",
       y = "Average Manhattan Distance", 
       caption = "Figure 1: Average Manhattan distance for single CpGs as a function of position relative to gene.") +
  scale_color_brewer(palette = "Set2") +
  scale_fill_brewer(palette = "Set2")

ggsave(filename = here("figures", "figure-1.png"), width = 7, dpi = 320, height = 4.326)
```


## Figure 2

```{r figure2}
permed_data %>%
  pivot_longer(dplyr::matches("1|3")) %>%
  separate(name, into = c("type", "method"), sep = "_v") %>%
  mutate(method = factor(method, c("1", "3"),
                             c("Naive method",
                               "Boot-strapping method")),
         type = factor(type, c("colon_pwd", "si_pwd", "endo_pwd"),
                             c("Colon", "Small Intestine", "Endometrium"))) %>%
  ggplot(aes(value, fill = type)) +
  geom_histogram(bins = 50, color = "grey30") +
  facet_grid(method~type, scales = "free") +
  theme_minimal() +
  labs(x = "p-value",
       y = "Number of genes",
       title = "Applying boot-strapping over the Naive method allows for \nbetter ranking of highly conserved genes",
       caption = "Figure 2: Distribution of boot-strapping p-values for genes as a function of tissue type and method used.")  +
  theme(legend.position = "none") +
  scale_color_brewer(palette = "Set2") +
  scale_fill_brewer(palette = "Set2")

ggsave(filename = here("figures", "figure-2.png"), width = 7, dpi = 320, height = 4.326)
```

## Figure 3

```{r figure3}
list(
  Colon             = enrich_colon$Description,
  `Small Intestine` = enrich_si$Description,
  Endometrium       = enrich_endo$Description
) %>%
  UpSetR::fromList() %>%
  UpSetR::upset(sets.bar.color = RColorBrewer::brewer.pal(3, "Set2")[c(2, 3, 1)])
grid::grid.text("Figure 3: Frequency of overlap between pathways that are called \nas most conserved for each tissue type.",x = 0.65, y=0.05, gp=grid::gpar(fontsize=10))

png(file=here("figures", "figure-3.png"), width = 7, height = 4.326, units = "in", res = 320) # or other device
list(
  Colon             = enrich_colon$Description,
  `Small Intestine` = enrich_si$Description,
  Endometrium       = enrich_endo$Description
) %>%
  UpSetR::fromList() %>%
  UpSetR::upset(sets.bar.color = RColorBrewer::brewer.pal(3, "Set2")[c(2, 3, 1)])
grid::grid.text("Figure 3: Frequency of overlap between pathways that are called \nas most conserved for each tissue type.",x = 0.65, y=0.05, gp=grid::gpar(fontsize=10))
dev.off()
```

## Figure 4

```{r figure4}
ggg <- enrich_si %>%
  ReactomePA::emapplot(showCategory = 30, layout = "graphopt") +
  labs(caption = "Figure 4: Relationship between pathways that are called as conserved in small intestine tissue.",
       title = "Relationship between pathways for 30 most over-represented pathways")

ggg$layers <- ggg$layers[1:2]

ggg + 
  ggraph::geom_node_label(aes_(label = ~name), repel = TRUE, size = 2) +
  scale_color_gradientn(colors = "black") +
  guides(color = "none",
         size = "none")

ggsave(filename = here("figures", "figure-4.png"), width = 7, dpi = 320, height = 4.326)
```



## Figure 5

```{r figure5}
figure5_data <- permed_data %>% 
  left_join(by = c("UCSC_RefGene_Name" = "gene"),
  promoted_data
  ) %>%
  left_join(expression_atlas %>% dplyr::select(`Gene Name`, colon, endometrium, `small intestine`),
            by = c("UCSC_RefGene_Name" = "Gene Name")) 

bind_rows(
  figure5_data %>%
  select(colon_pwd_v3, colon_pro, colon) %>%
  pivot_longer(-colon) %>%
  drop_na() %>%
  rename(expression = colon),
  figure5_data %>%
  select(si_pwd_v3, si_pro, `small intestine`) %>%
  pivot_longer(-`small intestine`) %>%
  drop_na() %>%
  rename(expression = `small intestine`),
  figure5_data %>%
  select(endo_pwd_v3, endo_pro, endometrium) %>%
  pivot_longer(-endometrium) %>%
  drop_na() %>%
  rename(expression = endometrium)
) %>%
  group_by(name) %>%
  mutate(value = factor(as.numeric(cut_number(value, 10)))) %>%
  separate(name, c("type", "method"), sep = "_", extra = "merge") %>%
  mutate(type = factor(type, c("colon", "si", "endo"), c("Colon", "Small Intenstine", "Endometrium")),
         method = factor(method, c("pwd_v3", "pro"), c("Boot-strapped values", "Promoter region"))) %>%
  ggplot(aes(value, expression, fill = type)) +
  geom_boxplot() +
  scale_y_log10() +
  guides(color = "none") +
  facet_grid(method ~ type) +
  labs(x = "bin", y = "Expression",
       title = "Boot-strapped values have higher correlation with expression \nthen just using Promoter regions",
       caption = "Figure 5: Relationship between conservation and expression according to tissue type.") +
  theme_minimal() +
  scale_fill_brewer(palette = "Set2") +
  theme(legend.position = "none")

ggsave(filename = here("figures", "figure-5.png"), width = 7, dpi = 320, height = 4.326)
```

